src/format.rs

```diff
@@ -532,3 +533,116 @@ pub struct CommandHistory {
     #[serde(flatten)]
     pub last_fetch: Option<FetchCommand>,
 }
+
+////////////////////////////////////////////////////////////////////////////////////
+//                                                                                //
+//                                                                                //
+//                                                                                //
+//                             <json report output>                               //
+//                                                                                //
+//                                                                                //
+//                                                                                //
+////////////////////////////////////////////////////////////////////////////////////
+
+/// cargo-vet's `--output-format=json` for `check` and `suggest` on:
+///
+/// * success
+/// * audit failure
+/// * violation conflicts
+///
+/// Other errors like i/o or supply-chain integrity issues will show
+/// up as miette-style json errors.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JsonReport {
+    #[serde(flatten)]
+    pub conclusion: JsonReportConclusion,
+}
+
+/// The conclusion of running `check` or `suggest`
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(tag = "conclusion")]
+pub enum JsonReportConclusion {
+    /// Success! Everything's Good.
+    #[serde(rename = "success")]
+    Success(JsonReportSuccess),
+    /// The violations and audits/exemptions are contradictory!
+    #[serde(rename = "fail (violation)")]
+    FailForViolationConflict(JsonReportFailForViolationConflict),
+    /// The audit failed, here's why and what to do.
+    #[serde(rename = "fail (vetting)")]
+    FailForVet(JsonReportFailForVet),
+}
+
+/// Success! Everything is audited!
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JsonReportSuccess {
+    /// These packages are fully vetted
+    pub vetted_fully: Vec<JsonPackage>,
+    /// These packages are partially vetted (some audits but relies on an `exemption`).
+    pub vetted_partially: Vec<JsonPackage>,
+    /// These packages are exempted
+    pub vetted_with_exemptions: Vec<JsonPackage>,
+}
+
+/// Failure! The violations and audits/exemptions are contradictory!
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct JsonReportFailForViolationConflict {
+    /// These packages have the following conflicts
+    pub violations: SortedMap<PackageAndVersion, Vec<ViolationConflict>>,
```

Comment:
This is existing, but the fact we expose the internal `ViolationConflict` struct isn't the best API here.

Could we perhaps put a comment on some of these suboptimal API parts noting that we probably want to change this part of the json output? Now that we're actually specifying it, and building more complex tools on it, I'm more afraid about people depending on the current janky format and making it tricky for us to improve it.